cpp/foxglove/include/foxglove/context.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <optional>
+#include <string>
+
+struct foxglove_context;
+
+namespace foxglove {
+
+typedef foxglove_context ContextInner;
+
+class Context final {
+  friend class McapWriterOptions;
+  friend class Channel;
+  friend class WebSocketServerOptions;
+
+public:
+  Context();
+
+  static Context get_default();
+
+private:
+  explicit Context(const foxglove_context* context);
+
+  inline const ContextInner* get_inner() const {
```

Comment:
nit: inline is implied here (2nd sentence: https://en.cppreference.com/w/cpp/language/inline)
```suggestion
  const ContextInner* get_inner() const {
```